src/main/java/spireMapOverhaul/zones/brokenSpace/relics/BrokenWingBoots.java

```diff
@@ -0,0 +1,63 @@
+package spireMapOverhaul.zones.brokenSpace.relics;
+
+import com.evacipated.cardcrawl.modthespire.lib.ByRef;
+import com.evacipated.cardcrawl.modthespire.lib.SpireInstrumentPatch;
+import com.evacipated.cardcrawl.modthespire.lib.SpirePatch2;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.map.DungeonMap;
+import com.megacrit.cardcrawl.relics.AbstractRelic;
+import com.megacrit.cardcrawl.relics.Sundial;
+import com.megacrit.cardcrawl.relics.WingBoots;
+import javassist.CannotCompileException;
+import javassist.expr.ExprEditor;
+import javassist.expr.FieldAccess;
+import javassist.expr.NewExpr;
+import spireMapOverhaul.BetterMapGenerator;
+import spireMapOverhaul.patches.BetterMapGenPatch;
+
+import static spireMapOverhaul.SpireAnniversary6Mod.makeID;
+
+public class BrokenWingBoots extends BrokenRelic {
```

Comment:
The new effect also has the potential to cause trouble. This code is doing some weird stuff and reading through it made me pretty sure there would be bugs. I do see that it works for most events and event mostly works for combat events, which is actually a very good starting point.

However, I tried out the following as my first test case:
* Set this to choose between a random event and Masked Bandits
* Picked Masked Bandits
* Won the fight
* Saved and reloaded

This resulted in a crash. You should be able to reproduce it yourself.

It may be that this code, janky as it is, is pretty close to working in all cases and I just happened to find the one thing that needs to be fixed. Then again, it may be that there are other lurking issues and we just haven't found them yet (especially when you consider modded events and other stuff mods can do). This kind of thing can be a maintainability challenge.

I'll let @erasels weigh in with his thoughts as well, since I can see arguments for both options (trying to test and fix this vs. deciding that it's not worth the potential issues).

P.S.: For what it's worth, there are working examples of how to do this kind of "pick between events", though the one I know of is a single event that does it arther than a relic.

Reply:
I just looked into that crash, it seems relatively easily fixable (famous last words). I'm going to dive into it a little more today.

Reply:
nevermind? fixed it.